ilastik/applets/counting/countingGui.py

```diff
@@ -1102,521 +1103,19 @@ def updateSum(self, *args, **kw):
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-    QColor(127.0, 0.0, 0.0, 255.0).rgba(),
-]
+@lru_cache(maxsize=1)
+def countingColorTable():
```

Comment:
Why keep this as a function instead of just computing it once and storing it as a module attribute? It shouldn't be that expensive to calculate...

Reply:
Ok I started with this parametrized and thought how cool it would be to have this more generic, and adapt the colors based on label colors and so on, but soon found that the changes proliferate quite a bit - so in the end I removed params and set the cache size to 1 :D. Not _really_ needed in that state, I agree.